### PR TITLE
[GridLayer] redraw tiles properly after changing maxNativeZoom

### LINKED ISSUE #6438: [GridLayer] redraw doesn't draw tiles properly after changing maxNativeZoom
- [X] I've looked at the [documentation](https://leafletjs.com/reference-1.3.4.html#gridlayer) to make sure the behavior is documented and expected
- [X] I'm sure this is a Leaflet code issue, not an issue with my own code nor with the framework I'm using (Angular)
- [X] I've searched through the issues to make sure it's not yet reported

**Steps to reproduce**
- Scroll in to level where data is not available. For example somewhere on AtlanticaOcean
- When get "Map data not available" picture set maxNativeZoom to smaller and call TileLayer.redraw() function.

**Expected behavior**
TileLayer redraw all tiles: 
image

**Current behavior**
TileLayer redraw partly: 
image

**Environment**
- Leaflet version: 1.3.4 
- Browser (with version): Chrome 70.0.3538.110 (Official Build) (64-bit)
- OS/Platform (with version): Win 10

### PR DESCRIPTION
**Pull request by cherniavskii**
This is blind attempt to fix #6438

### Before / After:

---

## Playground link:
https://plnkr.co/edit/ORkCquW4i2kIYLyOmcjl?p=preview

### CODE REVIEW

**Comment by cherniavskii** [comment]
Btw, running CI tests in Firefox really helped here - at some point tests passed in PhantomJS, but failed in Firefox

**Comment by cherniavskii** [comment]
I guess I'll wait for one more approval for this PR just to make sure it will not break anything

**Comment by ghybs** [comment]
Happy new year!

Here is the updated Plunker showing the PR in action:
https://plnkr.co/edit/sYllVqYkicUfolcAP2SW?p=preview

Would probably benefit from a test.

**Comment by cherniavskii** [comment]
@ghybs Happy New Year to you too!

Thanks for looking into this.
Yeah, this is a good case for regression test, will try to add one later.

File: src/layer/tile/GridLayer.js [hunk 1]
```diff
@@ -229,6 +229,8 @@ export var GridLayer = Layer.extend({
 	redraw: function () {
 		if (this._map) {
 			this._removeAllTiles();
+			this._tileZoom = this._clampZoom(this._map.getZoom());
+			this._updateLevels();
```
Comment (johnd0e) [inline src/layer/tile/GridLayer.js:233]:
```suggestion
			var tileZoom = this._clampZoom(this._map.getZoom());
			if (tileZoom !== this._tileZoom) {
				this._tileZoom = tileZoom;
				this._updateLevels();
			}
```

Comment (cherniavskii) [reply src/layer/tile/GridLayer.js:233]: Hi @johnd0e
I don't remember the context of this PR, but feel free to update it if there is space for improvements!

**Comment by Falke-Design** [comment]
@mourner @IvanSanchez this would be a easy thing to review & merge

**Review by mourner — APPROVED** [review]

**Pull request merged**